File: src/main/java/com/anfema/ionclient/archive/IonArchive.java
package com.anfema.ionclient.archive;

import com.anfema.ionclient.IonConfig;

import io.reactivex.Completable;


public interface IonArchive
{
	/**
	 * Download the archive file for collection (defined by {@link IonConfig}), which should make app usable in offline mode.
	 */
	Completable downloadArchive();
}

File: src/main/java/com/anfema/ionclient/archive/IonArchive.kt
package com.anfema.ionclient.archive

import io.reactivex.Completable

internal interface IonArchive {

    /**
     * Download the archive file for collection (defined by [com.anfema.ionclient.CollectionProperties]),
     * which makes app usable in offline mode.
     */
    fun downloadArchive(): Completable
}
